test/unit/logic/transaction_pool.js

```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
+				bundled: true,
+			};
+			transactionPool.queueTransaction.callsArgWith(1, transaction);
+			transactionPool.processUnconfirmedTransaction(transaction, true, res => {
+				expect(res).to.deep.eql(transaction);
+				done();
+			});
+		});
+
+		it('should fail to process and verify transaction', done => {
+			const transaction = {
+				id: '1234L',
+				type: transactionTypes.MULTI,
+				bundled: false,
+			};
+
+			const processVerifyTransactionStub = sinon
+				.stub()
+				.callsArgWith(2, 'Failed to process unconfirmed transaction');
+			TransactionPool.__set__(
+				'__private.processVerifyTransaction',
+				processVerifyTransactionStub
+			);
+
+			transactionPool.processUnconfirmedTransaction(transaction, true, res => {
+				expect(res).to.deep.eql('Failed to process unconfirmed transaction');
+				done();
+			});
+		});
+
+		it('should process and verify transaction', done => {
+			const transaction = {
+				id: '1235674L',
+				type: transactionTypes.MULTI,
+				bundled: false,
+			};
+
+			const processVerifyTransactionStub = sinon
+				.stub()
+				.callsArgWith(2, transaction);
+			TransactionPool.__set__(
+				'__private.processVerifyTransaction',
+				processVerifyTransactionStub
+			);
+			transactionPool.processUnconfirmedTransaction(transaction, true, res => {
+				expect(res.id).to.deep.eql(transaction.id);
+				done();
+			});
+		});
+	});
+
+	describe('receiveTransactions', () => {
+		it('should throw error when transaction in invalid', () => {
+			const invalidTransaction = {
+				id: '10431411423423423L',
+				type: transactionTypes.MULTI,
+			};
+			return expect(() => {
+				transactionPool.receiveTransactions(invalidTransaction, {});
+			}).to.throw();
+		});
+
+		it('should receive transaction already processed error when the transaction is processed before', done => {
+			const transaction = [
+				{ id: '10431411423423423L', type: transactionTypes.MULTI },
+			];
+			transactionPool.processUnconfirmedTransaction = sinon
+				.stub()
+				.callsArgWith(
+					2,
+					'Transaction is already processed: 10431411423423423L',
+					null
+				);
+
+			expect(
+				transactionPool.receiveTransactions(transaction, {}, (err, res) => {
+					expect(err).to.eql(
+						'Transaction is already processed: 10431411423423423L'
+					);
+					expect(res).to.deep.eql(transaction);
+					done();
+				})
+			);
+		});
+
+		it('should add transaction to queue when bundle is enabled', done => {
+			const transaction = [
+				{ id: '109249333874723L', type: transactionTypes.MULTI, bundled: true },
+			];
+			transactionPool.processUnconfirmedTransaction = sinon
+				.stub()
+				.callsArgWith(2, null, transaction);
+			expect(
+				transactionPool.receiveTransactions(transaction, {}, (err, res) => {
+					expect(err).to.be.null;
+					expect(res).to.deep.eql(transaction);
+					done();
+				})
+			);
+		});
+	});
+
+	describe('processBundled', () => {
+		let processVerifyTransaction;
+		beforeEach(done => {
+			resetStates();
```

Comment:
Full tested object should be restore with every `it`, initialization should be performed inside first `beforeEach`